### PR TITLE
Feature - Add NicBaseSelector Component

### PR DESCRIPTION
**Pull request by marchah**
## Feature

Create a dropdown list to let user select predefined nicotine strength and base.
Add a custom option to let user enter it's own value

## Example of valid results

Screen Shot 2019-08-04 at 11 58 33 AM

### CODE REVIEW

File: src/NicBaseSelector.re [hunk 1]
```diff
@@ -0,0 +1,134 @@
+type nic = {
+  concentration: int,
+  base: Constants.base,
+  index: int,
+};
+
+[@react.component]
+let make = (~handleSetNicWeight) => {
+  let nicOptions: list(nic) = [
+    {concentration: 100, base: Constants.PG, index: 0},
```
Comment (marchah) [inline src/NicBaseSelector.re]: `index` is a hack, i didn't find a way to get the index from `List.map`

Comment (mlms13) [reply src/NicBaseSelector.re]: You can use `List.mapi`, which passes the index as the first argument to your map function: https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html#VALmapi

File: src/NicBaseSelector.re [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                     customNicBase,
+                   )
+               )
+             }
+           />
+           <select
+             name="nic-custom-selector"
+             onChange={event =>
+               setCustomNicBase(ReactEvent.Form.target(event)##value)
+               |> (
```
Comment (mlms13) [inline src/NicBaseSelector.re]: Does it work to read the `##value` from the event and pass it into `setCustomNicBase` here? I think the set function wants a typed value and sadly I think `##value` breaks out of the type system (but is actually a string). You may need to run it through `Constants.convertStringToBaseType` first.

Also, you don't need to pipe the result of this function into the next line. You could instead do:

```reason
onChange ={ event => {
  setCustomNicBase(...);
  handleSetNicWeight(...);
}}
```

Comment (marchah) [reply src/NicBaseSelector.re]: you're right i had a issue with the type thanks

**Pull request merged**